node/heartbeat/monitor.go

```diff
@@ -281,6 +293,22 @@ func (m *Monitor) computeShardID(pubkey string) uint32 {
 	return m.heartbeatMessages[pubkey].computedShardID
 }
 
+func (m *Monitor) isInEligibleList(pubkey []byte, shardID uint32) bool {
+	nodesMap, err := m.eligibleListProvider.GetNodesPerShard(m.epochHandler.Epoch())
+	if err != nil {
+		return false
```

Comment:
I would also add a log.warn(err) here, if something went wrong with the eligibleListProvier

Reply:
done